e2e_test/source_inline/kafka/avro/alter_table.slt

```diff
@@ -45,7 +45,28 @@ select * from t
 system ok
 sr_register avro_alter_table_test-value AVRO <<< '{"type":"record","name":"Root","fields":[{"name":"bar","type":"int","default":0},{"name":"foo","type":"string"},{"name":"nested","type":["null",{"type":"record","name":"Nested","fields":[{"name":"baz","type":"int"}]}],"default":null}]}'
 
-# Refresh table schema should succeed
+# Before refreshing schema, we create a `SINK INTO TABLE` which involves table replacement,
+# showing that this will NOT accidentally refresh the schema.
```

Comment:
Showing that

> 2\. For `SINK INTO TABLE`, ensure schema unchanged by using the defined schema after purification.
>
> Case 2 actually fixes a long-standing issue caused by impurity. Previously, creating a sink into a table will accidentally refresh its schema.

This will panic the compute node in the current release.